customMathJSImplementation.js

```diff
@@ -0,0 +1,444 @@
+    if (!responseValue) return false;
+
+    const baseID = match[1];
+    const cellInputValue = document.getElementById(id)
+      ?.form
+      ?.querySelector(`table tbody tr[data-question-id="${baseID}"] td[data-question-id="${baseID}"] input[id="${id}"]`)
+      ?.value;
+
+    if (!cellInputValue) {
+      console.error('CELL INPUT VALUE NOT FOUND:', id);
+      return false;
+    }
+
+    return cellInputValue === responseValue;
+  },
+
+  someSelected: function (...ids) { // TODO: test. this is impacted by the isSelected changes.
+    return (ids.some(id => this.isSelected(id)))
+  },
+
+  noneSelected: function(...ids){
+    return (!ids.some(id => this.isSelected(id)))
+  },
+  // defaultValue accepts an Id and a value or a Id/Value
+  // If only 1 default value is given, first it looks it up
+  // if it does not exist assume it is a value...
+  // If 2 default values are given, look up the first, if it
+  // does not exist, return the second as a value...
+  valueOrDefault: function (x, ...defaultValue) {
+    let v = this._value(x)
+
+    let indx = 0;
+    while (v == null && defaultValue.length > indx) {
+      v = this._value(defaultValue[indx])
+      if (v == null) indx++
+    }
+    
+    if (v == null) v = defaultValue[defaultValue.length - 1]
+
+    return (v);
+  },
+
+  selectionCount: function(x, countReset=false) {
+    let [questionId, name] = x.split(':')
+    name = name ?? questionId
+
+    if (!this.exists(questionId)) return 0
+    let responseValue = this._value(questionId);
+
+    if (Array.isArray(responseValue) || Array.isArray(responseValue[name])) {
+      responseValue = Array.isArray(responseValue) ? responseValue : responseValue[name]
+
+      if (countReset){
+        return responseValue.length;
+      }
+
+      console.warn('TODO: (selectionCount) remove DOM access and use stateManager', x);
+
+      // BUG FIX:  if the data-reset ("none of the above") is selected
+      let questionElement = document.getElementById(questionId)
+      // there is a chance that nothing is selected (v.length==0) in that case you will the 
+      // selector will find nothing.  Use the "?" because you cannot find the dataset on a null object.
+      return questionElement.querySelector(`input[type="checkbox"][name="${name}"]:checked`)?.dataset["reset"]?0:responseValue.length
+    }
+
+    // if we want object to return the number of keys
+    // Object.keys(v).length
+    // otherwise:
+    return 0;
+  },
+  // // For a question in a loop, does the value of the response
+  // // for ANY ITERATION equal a value from a given set. 
+  // loopQuestionValueIsOneOf: function (id, ...values) {
+  //   // Loops append _n_n to the id, where n is an
+  //   // integer starting from 1...
+  //   for (let i = 1; ; i = i + 1) {
+  //     let tmp_qid = `${id}_${i}_${i}`
+  //     // the Id does not exist, we've gone through
+  //     // all potential question and have not found
```

Comment:
intentionally commented out?

Reply:
Yes, and I wasn't quite ready to get rid of it. I haven't found any refs to `loopQuestionValueIsOneOf` or `gridQuestionValueIsOneOf` in this codebase or the questionnaire repo, so I think it may be unused. I'd like to keep it for now and remove it in the cleanup pass later when I'm sure I haven't missed a use case.